### PR TITLE
Add missing guards to profiling calls

### PR DESCRIPTION
**Pull request by letmaik**
**Description**: `Profiler::StartTime()` is now only called in `InferenceSession` when profiling is enabled. An extra `ORT_ENFORCE` in the profiler enforces this now.

**Motivation and Context**
- Why is this change required? What problem does it solve?
Other code already has guards around those calls, so this PR just makes it consistent.

### CODE REVIEW

**Comment by letmaik** [comment]
Is there any reason why this PR is being ignored?
@snnn

File: onnxruntime/core/session/inference_session.cc [hunk 1]
```diff
@@ -186,7 +186,10 @@ common::Status InferenceSession::RegisterCustomRegistry(std::shared_ptr<CustomRe
 
 common::Status InferenceSession::Load(std::function<common::Status(std::shared_ptr<Model>&)> loader, const std::string& event_name) {
   Status status = Status::OK();
-  auto tp = session_profiler_.StartTime();
+  TimePoint tp;
+  if (session_profiler_.FEnabled()) {
```
Comment (skottmckay) [inline onnxruntime/core/session/inference_session.cc]: Should this be IsEnabled() not FEnabled()?

Comment (letmaik) [reply onnxruntime/core/session/inference_session.cc]: When I submitted the PR the method name was `FEnabled`. In the meantime it changed to `IsEnabled`. One more reason to do quick PR reviews :) I'll fix it and merge in the latest master.

**Pull request merged**